### PR TITLE
chore(bottlecap): add `hyper@0.x` in telemetry api

### PR DESCRIPTION
**Pull request by duncanista**
# What?

Use `hyper` crate instead of parsing payload for the telemetry API.

# Motivation

We should have done this since the start, but we weren't using tokio

### CODE REVIEW

File: bottlecap/src/bin/bottlecap/main.rs [hunk 1]
```diff
@@ -538,22 +538,28 @@ async fn start_dogstatsd(
...
+    //     &telemetry_listener_config,
+    //     logs_agent_channel,
+    //     telemetry_listener_cancel_token.clone(),
+    // )
+    // .await;
+    // // .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))?;
+    // // tokio::spawn(async move {
+    // //     telemetry_listener.spin().await;
+    // // });
+    let ct_clone = telemetry_listener_cancel_token.clone();
```
Comment (duncanista) [inline bottlecap/src/bin/bottlecap/main.rs:558]: I'm gonna remove this comment

File: bottlecap/src/bin/bottlecap/main.rs [hunk 2]
```diff
@@ -538,22 +538,28 @@ async fn start_dogstatsd(
...
     tokio::spawn(async move {
-        telemetry_listener.spin().await;
+        let _telemetry_listener =
+            TelemetryListener::new_hyper(&telemetry_listener_config, logs_agent_channel, ct_clone)
```
Comment (duncanista) [inline bottlecap/src/bin/bottlecap/main.rs:561]: I'm renaming this to `new`

Comment (duncanista) [reply bottlecap/src/bin/bottlecap/main.rs]: Renamed to `spin`, `new` convention returns a Self most of the time

**Review by duncanista — COMMENTED** [review]
Ready to review

**Review by astuyve — APPROVED** [review]
area man approves of his own commits

**Pull request merged**